services/services.go

```diff
@@ -53,8 +49,29 @@ type Service interface {
 	SetSharding(s ServiceSharding) Service
 }
 
-// NewService creates a new Service
-func NewService() Service { return new(service) }
+// PlacementSnapshot describes how shards are placed on instances
+type PlacementSnapshot interface {
+	// Instances returns the placement instances
+	Instances() []PlacementInstance
+
+	// SetInstances sets the placement instances
+	SetInstances(insts []PlacementInstance) PlacementSnapshot
+
+	// Replication returns the placement replication description or nil if none
+	Replication() ServiceReplication
+
+	// SetReplication sets the placement replication description or nil if none
+	SetReplication(r ServiceReplication) PlacementSnapshot
+
+	// Sharding returns the placement sharding description or nil if none
+	Sharding() ServiceSharding
+
+	// SetSharding sets the placement sharding description or nil if none
+	SetSharding(s ServiceSharding) PlacementSnapshot
+
+	// Validate checks if the placement is valid
+	Validate() error
```

Comment:
Do we need `Validate()` anymore if this is just a plain old object or POGO one might say?